File: Recast/iTunesPodcasts/Models/iTunes/iTunesNamespace.m
//
//  iTunesNamespace.m
//  Recast
//
//  Created by Mindy Lou on 10/21/18.
//  Copyright © 2018 Cornell AppDev. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "iTunesNamespace.h"

@implementation ITunesNamespace

@dynamic author;
@dynamic block;
@dynamic categories;
@dynamic image;
@dynamic duration;
@dynamic explicit;
@dynamic isClosedCaptioned;
@dynamic order;
@dynamic complete;
@dynamic theNewFeedUrl;
@dynamic owner;
@dynamic subtitle;
@dynamic summary;
@dynamic keywords;
@dynamic type;
@dynamic episodeType;
@dynamic season;
@dynamic episode;

+ (PodcastType)initPodcastTypeWithRawValue:(NSString *)rawValue {
    if ([rawValue isEqualToString: @"episodic"]) {
        return episodic;
    } else if ([rawValue isEqualToString:@"serial"]) {
        return serial;
    } else if ([rawValue isEqualToString:@"episodic"]) {
        return episodic;
    }
    return NULL;
}

+ (EpisodeType)initEpisodeTypeWithRawValue:(NSString *)rawValue {
    if ([rawValue isEqualToString:@"full"]) {
        return full;
    } else if ([rawValue isEqualToString:@"trailer"]) {
        return trailer;
    } else if ([rawValue isEqualToString:@"bonus"]) {
        return bonus;
    }
    return NULL;
}

@end

File: Recast/iTunesPodcasts/Models/iTunes/iTunesNamespace.swift
//
//  iTunesNamespace.swift
//
//  Copyright (c) 2017 Ben Murphy
//
//  Permission is hereby granted, free of charge, to any person obtaining a copy
//  of this software and associated documentation files (the "Software"), to deal
//  in the Software without restriction, including without limitation the rights
//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
//  copies of the Software, and to permit persons to whom the Software is
//  furnished to do so, subject to the following conditions:
//
//  The above copyright notice and this permission notice shall be included in all
//  copies or substantial portions of the Software.
//
//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
//  SOFTWARE.
//

import Foundation

// MARK: - Equatable

extension ITunesNamespace {

    public static func == (lhs: ITunesNamespace, rhs: ITunesNamespace) -> Bool {
        return
            lhs.author == rhs.author &&
            lhs.block == rhs.block &&
            lhs.image == rhs.image &&
            lhs.duration == rhs.duration &&
            lhs.explicit == rhs.explicit &&
            lhs.isClosedCaptioned == rhs.isClosedCaptioned &&
            lhs.complete == rhs.complete &&
            lhs.newFeedUrl == rhs.newFeedUrl &&
            lhs.owner == rhs.owner &&
            lhs.subtitle == rhs.subtitle &&
            lhs.summary == rhs.summary &&
            lhs.keywords == rhs.keywords &&
            lhs.podcastType == rhs.podcastType &&
            lhs.episodeType == rhs.episodeType &&
            lhs.season == rhs.season &&
            lhs.episode == rhs.episode
    }

}
